### ISSUE TITLE
Error: No file in /home/runner/work/ansible/ansible matched to [~/requirements.txt or **/pyproject.toml]

### ISSUE THREAD

**Issue by 0lzi** [ansible/ansible-lint#3648]
##### Summary

Using the GH action with the following .yml

```
---
name: ansible-lint
on:
  push:
    branches:
      - '*'
  pull_request:
    branches: ["main"]
jobs:
  build:
    name: Ansible Lint
    runs-on: ubuntu-latest
    steps:
      - name: Run ansible-lint
        uses: ansible/ansible-lint@v6.17.2
```

I get the following Error when running

`Error: No file in /home/runner/work/ansible/ansible matched to [~/requirements.txt or **/pyproject.toml], make sure you have checked out the target repository`

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

Github Actions

##### STEPS TO REPRODUCE

Create `ansible-lint.yml` in `.github/workflows` folder

pasted 

```
---
name: ansible-lint
on:
  push:
    branches:
      - '*'
  pull_request:
    branches: ["main"]
jobs:
  build:
    name: Ansible Lint
    runs-on: ubuntu-latest
    steps:
      - name: Run ansible-lint
        uses: ansible/ansible-lint@v6.17.2
```
Push to new branch,get below error

```
Error: No file in /home/runner/work/ansible/ansible matched to [~/requirements.txt or **/pyproject.toml], make sure you have checked out the target repository
```

##### Desired Behavior

The action workflow runs ansible-lint on repository

##### Actual Behavior

Workflow fails

**Comment by ReenigneArcher** [comment]
I'm getting the same error after changed from the archived action.

But I do have a requirements.txt in the root of my repo and have checked out the repo.

```yml
jobs:
  ansible_lint:
    name: ansible lint
    runs-on: [self-hosted, linux, X64]
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Ansible lint
        id: ansible-lint
        uses: ansible/ansible-lint@v6.17.2
        with:
          path: "playbooks/"
```

**Comment by benjy44** [comment]
I experience the same issue
Running ACT locally does not produce any issue
Running previous version `v6.17.1` causes a completely different issue about `× Getting requirements to build wheel did not run successfully.`

**Comment by alinabuzachis** [comment]
We are also facing this issue: https://github.com/redhat-cop/cloud.aws_ops/pull/84 and https://github.com/redhat-cop/cloud.aws_troubleshooting/pull/24

**Comment by AlexTMjugador** [comment]
I was affected by this and found out that this problem has already been silently addressed. The `v6.17.2` tag was last modified minutes before the commit that fixes this issue was pushed: https://github.com/ansible/ansible-lint/commit/6f728e0c95671350595c14f46fe71738adc23d2e

Therefore, the `v6.17.2` tag is known to be broken. Until a new tag with that commit is pushed, workflows should either:
- Use `ansible/ansible-lint@main`, as mentioned in the `README.md` file.
- Pin their `ansible-lint` action dependency to the aforementioned commit, like this: `ansible/ansible-lint@6f728e0c95671350595c14f46fe71738adc23d2e`.

**Comment by ssbarnea** [comment]
Sadly I am unable to just remove that version but I added a comment, recommending everyone to use `v6.18.0`.
